### ISSUE TITLE
WordPress.DateTime.RestrictedFunctions.date_date - only flag when second parameter is missing

### ISSUE THREAD

**Issue by rebeccahum** [WordPress/WordPress-Coding-Standards#1972]
For `WordPress.DateTime.RestrictedFunctions.date_date`, it flags the below as an error with the message:
 
> date() is affected by runtime timezone changes which can cause date/time to be incorrectly displayed. Use gmdate() instead.

However, when there's a specific timestamp being passed in, does the above still apply?

```
$date = date( PREFIXED_DATE_TIME_FORMAT, $data['details']['publish_timestamp'] );
```

I figured that it would only flag for these cases:

```
$date = date();
$date = date( PREFIXED_DATE_TIME_FORMAT );
$date = date( PREFIXED_DATE_TIME_FORMAT, null );
```

**Comment by WickedWolfy** [comment]
Reasoning:

The issue with the timezones is understood, but, as developers, I think it is our responsibility, not the bots.
Warning is always welcome - an alert that "some values might not be proper".

Also, in case of having timestamp, the timezone is not applied, so there should not even be a warning.

Thanks.

**Comment by jrfnl** [comment]
@Rarst Care to chime in, please ?

**Comment by Rarst** [comment]
> However, when there's a specific timestamp being passed in, does the above still apply?

Yes, because output is affected as well:

```php
$now = time();
var_dump( date( DATE_W3C, $now ) ); // 2021-02-01T19:47:50+00:00
date_default_timezone_set( 'Europe/Kiev' );
var_dump( date( DATE_W3C, $now ) ); // 2021-02-01T21:47:50+02:00
```

> Also, in case of having timestamp, the timezone is not applied, so there should not even be a warning.

`date()` always applies default time zone to the output.

As far as _core development_ is concerned - any use of `date()` is an error and `gmdate()` is a suggested safe replacement.

Outside of core you can always disable the sniff. Personally I wouldn't recommend.
